modules/mobile-agents/windows/jax-rs/src/main/java/org/wso2/carbon/mdm/mobileservices/windows/operations/util/OperationReply.java

```diff
@@ -222,22 +343,314 @@ private Item appendGetInfo(Operation operation) {
+                        item.setData("0");
+                    }
+                }
+                if (operation.getCode().equals(org.wso2.carbon.mdm.mobileservices.windows.common.Constants
+                        .OperationCodes.ENCRYPT_STORAGE)) {
+                    JSONObject payload = new JSONObject(operation.getPayLoad().toString());
+                    if (payload.getBoolean("encrypted")) {
+                        Meta meta = new Meta();
+                        meta.setFormat("int");
+                        item.setTarget(target);
+                        item.setMeta(meta);
+                        item.setData("1");
+                    } else {
+                        Meta meta = new Meta();
+                        meta.setFormat("int");
+                        item.setTarget(target);
+                        item.setMeta(meta);
+                        item.setData("0");
+                    }
+
+                }
+            }
+        }
+        return item;
+    }
+
+    private List<Add> appendAddInfo(Operation operation) throws WindowsOperationException {
+
+        List<Add> addList = new ArrayList<>();
+
+        Gson gson = new Gson();
+        String operationCode = operation.getCode();
+        if (operationCode.equals("PASSCODE_POLICY")) {
+            PasscodePolicy passcodeObject = gson.fromJson((String) operation.getPayLoad(), PasscodePolicy.class);
+            for (Configure configure : Configure.values()) {
+                String maxAttempt = "PASSWORD_MAX_FAIL_ATTEMPTS";
+                if (operationCode != null && maxAttempt.equals(configure.name())) {
+                    int maxFailedAttempts = passcodeObject.getMaxFailedAttempts();
+                    String attempt = String.valueOf(maxFailedAttempts);
+                    Add add = new Add();
+                    List<Item> itemList = new ArrayList<>();
+                    Item item = new Item();
+                    Target target = new Target();
+                    target.setLocURI(configure.getCode());
+                    Meta meta = new Meta();
+                    meta.setFormat("int");
+                    item.setTarget(target);
+                    item.setMeta(meta);
+                    item.setData(attempt);
+                    itemList.add(item);
+                    add.setCommandId(90);
+                    add.setItems(itemList);
+                    addList.add(add);
+                }
+                String allowPassword = "DEVICE_PASSWORD_ENABLE";
+                if (operationCode != null && allowPassword.equals(configure.name())) {
+                    List<Item> itemList = new ArrayList<>();
+                    if (passcodeObject.isEnablePassword()) {
+                        Add add = new Add();
+                        Item item = new Item();
+                        Target target = new Target();
+                        target.setLocURI(configure.getCode());
+                        Meta meta = new Meta();
```

Comment:
The same logic, such as, creating "Meta", "Add", etc instances, seems to be repeated in many places. Might be a good idea to introduce a few util methods wrapping up the instantiation of each of those beans to make the code more readable and organized.